### PR TITLE
Linux platform refactoring

### PR DESCRIPTION
**Pull request by yan**
* Move stack-based helpers to `Cpu`
 * Refactor initialization so that we can create a platform model that isn't associated with an ELF file
 * Simplify SLinux sys call handling to rely on normal `ConcretizeArgument` exceptions

### CODE REVIEW

File: manticore/core/cpu/abstractcpu.py [hunk 1]
```diff
@@ -477,6 +478,74 @@ def read_bytes(self, where, size):
...
+            where += 1
+        return s.getvalue()
+
+    def push_bytes(self, data):
+        '''
+        Write `data` to the stack and decrement the stack pointer accordingly.
+
+        :param str data: Data to write
+        '''
+        self.STACK -= len(data)
```
Comment (feliam) [inline manticore/core/cpu/abstractcpu.py]: Should we assert  that ` 'STACK' in self._regfile`  at class constructor?

File: manticore/core/cpu/abstractcpu.py [hunk 2]
```diff
@@ -477,6 +478,74 @@ def read_bytes(self, where, size):
...
+        :param int where: Address to read string from
+        :param int max_length:
+            The size in bytes to cap the string at, or None [default] for no
+            limit.
+        :return: string read
+        :rtype: str
+        '''
+        s = StringIO.StringIO()
+        while True:
+            c = self.read_int(where, 8)
```
Comment (feliam) [inline manticore/core/cpu/abstractcpu.py]: What if c is symbolic?

File: manticore/core/cpu/abstractcpu.py (hunk 1 above)
Comment (yan) [inline manticore/core/cpu/abstractcpu.py]: do we not assume that `STACK` is a universal alias?

File: manticore/core/cpu/abstractcpu.py (hunk 2 above)
Comment (yan) [inline manticore/core/cpu/abstractcpu.py]: This function is technically defined to work solely on concrete buffers. What would be the correct behavior? Implement it via Mark's `find_zero`?

File: manticore/core/cpu/abstractcpu.py (hunk 1 above)
Comment (feliam) [inline manticore/core/cpu/abstractcpu.py]: So aliases are passed to RegisterFile constructor and those may not contain 'STACK' as it is now. If we want STACK to be part of RegisterFile I say we should assert that an specific alias for it is provided. This is at RegisterFile constructor or AbstractCpu I guess.  Same for 'PC'. No?

File: manticore/core/cpu/abstractcpu.py (hunk 2 above)
Comment (feliam) [inline manticore/core/cpu/abstractcpu.py]: If we are sure it's going to be concrete then just assert not issymbolic(x).

File: manticore/core/cpu/abstractcpu.py (hunk 1 above)
Comment (yan) [inline manticore/core/cpu/abstractcpu.py]: Added alias check to the `Cpu` ctor

File: manticore/core/cpu/abstractcpu.py (hunk 2 above)
Comment (yan) [inline manticore/core/cpu/abstractcpu.py]: 👍

**Comment by yan** [comment]
@feliam still approved with the above changes?

**Pull request merged**